File: lua/str_join.lua
local chunk = "regexp123abcbcd"
local repeatCount = 10000
local size = 1000

function join()
  chunks = {}
  chunks[size] = chunk
  for i = 1, size - 1 do
    chunks[i] = chunk
  end
  table.concat(chunks, "")
end

local start = os.clock()
for i = 1, repeatCount do
  join()
end
local _end = os.clock()

print((_end - start) * 1000)

File: ruby/str_join.rb
chunk = "regexp123abcbcd"
repeat_count = 10000
size = 1000

def join(size, chunk)
  Array.new(size).map { chunk }.join("")
end

start = Time.now
repeat_count.times { join(size, chunk) }
_end = Time.now

puts ((_end - start) * 1000).to_i
